numba/cpython/randomimpl.py

```diff
@@ -52,6 +52,20 @@ def const_int(x):
 rnd_state_ptr_t = ir.PointerType(rnd_state_t)
 
 
+def _is_valid_size(size):
```

Comment:
There's a function with similar functionality. Maybe it could be reused.

https://github.com/numba/numba/blob/3e96339a523b3cf4b737b8fff7dba6ee64927584/numba/np/random/generator_methods.py#L59

Otherwise, The current implementation, while being efficient might be hard to read straightaway, and would be better if it simply was a bunch of conditions:

```python
if isinstance(size, types.Integer) or (isinstance(size, types.UniTuple) and isinstance(size.dtype, types.Integer)).....:
    return True
```